modules/serdes/include/hephaestus/serdes/protobuf/enums.h

```diff
@@ -0,0 +1,110 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <string_view>
+#include <type_traits>
+#include <unordered_map>
+
+#include <fmt/core.h>
+#include <magic_enum.hpp>
+
+#include "hephaestus/utils/concepts.h"
+#include "hephaestus/utils/exception.h"
+#include "hephaestus/utils/format/format.h"
+#include "hephaestus/utils/string/string_literal.h"
+#include "hephaestus/utils/string/string_utils.h"
+
+namespace heph::serdes::protobuf {
+
+template <EnumType ProtoT, EnumType T>
+[[nodiscard]] auto toProtoEnum(const T& enum_value) -> ProtoT;
+
+template <EnumType ProtoT, EnumType T>
+auto fromProto(const ProtoT& proto_enum_value, T& enum_value) -> void;
```

Comment:
I wonder if it is not better to be consistent with the `toProtoEnum` and have `fromProtoEnum` and make it return the value.

Reply:
Your call, both is fine with me. You will however have to specify the non proto type as a template parameter then.

Reply:
ahh right...ok let's leave it as it is